diff --git a/public/admin/tool/xmldb/actions/view_structure_sql/view_structure_sql.class.php b/public/admin/tool/xmldb/actions/view_structure_sql/view_structure_sql.class.php
index 3a3324c..169903d 100644
--- a/public/admin/tool/xmldb/actions/view_structure_sql/view_structure_sql.class.php
+++ b/public/admin/tool/xmldb/actions/view_structure_sql/view_structure_sql.class.php
@@ -81,7 +81,7 @@ class view_structure_sql extends XMLDBAction {
 
         // The back to edit table button
         $b = ' <p class="centerpara buttons">';
-        $b .= '<a href="index.php?action=edit_xml_file&amp;dir=' . urlencode(str_replace($CFG->dirroot, '', $dirpath)) . '">[' . $this->str['back'] . ']</a>';
+        $b .= '<a href="index.php?action=edit_xml_file&amp;dir=' . urlencode(\core\component::to_mono_path($dirpath, true)) . '">[' . $this->str['back'] . ']</a>';
         $b .= '</p>';
         $o = $b;
 
