diff --git a/public/admin/tool/xmldb/actions/edit_key_save/edit_key_save.class.php b/public/admin/tool/xmldb/actions/edit_key_save/edit_key_save.class.php
index 7dd00e7..6a91b86 100644
--- a/public/admin/tool/xmldb/actions/edit_key_save/edit_key_save.class.php
+++ b/public/admin/tool/xmldb/actions/edit_key_save/edit_key_save.class.php
@@ -245,7 +245,7 @@ class edit_key_save extends XMLDBAction {
             $o = '<p>' .implode(', ', $errors) . '</p>
                   <p>' . $name . ': ' . $tempkey->readableInfo() . '</p>';
             $o.= '<a href="index.php?action=edit_key&amp;key=' .$key->getName() . '&amp;table=' . $table->getName() .
-                 '&amp;dir=' . urlencode(str_replace($CFG->dirroot, '', $dirpath)) . '">[' . $this->str['back'] . ']</a>';
+                 '&amp;dir=' . urlencode(\core\component::to_mono_path($dirpath, true)) . '">[' . $this->str['back'] . ']</a>';
             $this->output = $o;
         }
 
