diff --git a/public/admin/tool/xmldb/actions/edit_key/edit_key.class.php b/public/admin/tool/xmldb/actions/edit_key/edit_key.class.php
index 27a5c9e..c793193 100644
--- a/public/admin/tool/xmldb/actions/edit_key/edit_key.class.php
+++ b/public/admin/tool/xmldb/actions/edit_key/edit_key.class.php
@@ -100,7 +100,7 @@ class edit_key extends XMLDBAction {
         // Add the main form
         $o = '<form id="form" action="index.php" method="post">';
         $o.= '<div>';
-        $o.= '    <input type="hidden" name ="dir" value="' . str_replace($CFG->dirroot, '', $dirpath) . '" />';
+        $o.= '    <input type="hidden" name ="dir" value="' . \core\component::to_mono_path($dirpath, true) . '" />';
         $o.= '    <input type="hidden" name ="table" value="' . $tableparam .'" />';
         $o.= '    <input type="hidden" name ="key" value="' . $keyparam .'" />';
         $o.= '    <input type="hidden" name ="sesskey" value="' . sesskey() .'" />';
@@ -149,18 +149,18 @@ class edit_key extends XMLDBAction {
         $b = ' <p class="centerpara buttons">';
         // The view original XML button
         if ($table->getKey($keyparam)) {
-            $b .= '&nbsp;<a href="index.php?action=view_key_xml&amp;dir=' . urlencode(str_replace($CFG->dirroot, '', $dirpath)) . '&amp;select=original&amp;table=' . $tableparam . '&amp;key=' . $keyparam . '">[' . $this->str['vieworiginal'] . ']</a>';
+            $b .= '&nbsp;<a href="index.php?action=view_key_xml&amp;dir=' . urlencode(\core\component::to_mono_path($dirpath, true)) . '&amp;select=original&amp;table=' . $tableparam . '&amp;key=' . $keyparam . '">[' . $this->str['vieworiginal'] . ']</a>';
         } else {
             $b .= '&nbsp;[' . $this->str['vieworiginal'] . ']';
         }
         // The view edited XML button
         if ($key->hasChanged()) {
-            $b .= '&nbsp;<a href="index.php?action=view_key_xml&amp;dir=' . urlencode(str_replace($CFG->dirroot, '', $dirpath)) . '&amp;select=edited&amp;table=' . $tableparam . '&amp;key=' . $keyparam . '">[' . $this->str['viewedited'] . ']</a>';
+            $b .= '&nbsp;<a href="index.php?action=view_key_xml&amp;dir=' . urlencode(\core\component::to_mono_path($dirpath, true)) . '&amp;select=edited&amp;table=' . $tableparam . '&amp;key=' . $keyparam . '">[' . $this->str['viewedited'] . ']</a>';
         } else {
             $b .= '&nbsp;[' . $this->str['viewedited'] . ']';
         }
         // The back to edit table button
-        $b .= '&nbsp;<a href="index.php?action=edit_table&amp;table=' . $tableparam . '&amp;dir=' . urlencode(str_replace($CFG->dirroot, '', $dirpath)) . '">[' . $this->str['back'] . ']</a>';
+        $b .= '&nbsp;<a href="index.php?action=edit_table&amp;table=' . $tableparam . '&amp;dir=' . urlencode(\core\component::to_mono_path($dirpath, true)) . '">[' . $this->str['back'] . ']</a>';
         $b .= '</p>';
         $o .= $b;
 
