diff --git a/public/admin/tool/xmldb/actions/edit_table_save/edit_table_save.class.php b/public/admin/tool/xmldb/actions/edit_table_save/edit_table_save.class.php
index 1332570..6b1560a 100644
--- a/public/admin/tool/xmldb/actions/edit_table_save/edit_table_save.class.php
+++ b/public/admin/tool/xmldb/actions/edit_table_save/edit_table_save.class.php
@@ -107,7 +107,7 @@ class edit_table_save extends XMLDBAction {
             $o = '<p>' .implode(', ', $errors) . '</p>
                   <p>' . $temptable->getName() . '</p>';
             $o.= '<a href="index.php?action=edit_table&amp;table=' . $tableparam .
-                 '&amp;dir=' . urlencode(str_replace($CFG->dirroot, '', $dirpath)) . '">[' . $this->str['back'] . ']</a>';
+                 '&amp;dir=' . urlencode(\core\component::to_mono_path($dirpath, true)) . '">[' . $this->str['back'] . ']</a>';
             $this->output = $o;
 
 
