diff --git a/public/admin/tool/xmldb/actions/edit_field_save/edit_field_save.class.php b/public/admin/tool/xmldb/actions/edit_field_save/edit_field_save.class.php
index 67ddf04..e20673d 100644
--- a/public/admin/tool/xmldb/actions/edit_field_save/edit_field_save.class.php
+++ b/public/admin/tool/xmldb/actions/edit_field_save/edit_field_save.class.php
@@ -216,7 +216,7 @@ class edit_field_save extends XMLDBAction {
             $o = '<p>' .implode(', ', $errors) . '</p>
                   <p>' . $name . ': ' . $tempfield->readableInfo() . '</p>';
             $o.= '<a href="index.php?action=edit_field&amp;field=' . $field->getName() . '&amp;table=' . $table->getName() .
-                 '&amp;dir=' . urlencode(str_replace($CFG->dirroot, '', $dirpath)) . '">[' . $this->str['back'] . ']</a>';
+                 '&amp;dir=' . urlencode(\core\component::to_mono_path($dirpath, true)) . '">[' . $this->str['back'] . ']</a>';
             $this->output = $o;
         }
 
