diff --git a/public/admin/tool/xmldb/actions/edit_index/edit_index.class.php b/public/admin/tool/xmldb/actions/edit_index/edit_index.class.php
index c88505f..f715e60 100644
--- a/public/admin/tool/xmldb/actions/edit_index/edit_index.class.php
+++ b/public/admin/tool/xmldb/actions/edit_index/edit_index.class.php
@@ -100,7 +100,7 @@ class edit_index extends XMLDBAction {
         // Add the main form
         $o = '<form id="form" action="index.php" method="post">';
         $o.= '<div>';
-        $o.= '    <input type="hidden" name ="dir" value="' . str_replace($CFG->dirroot, '', $dirpath) . '" />';
+        $o.= '    <input type="hidden" name ="dir" value="' . \core\component::to_mono_path($dirpath, true) . '" />';
         $o.= '    <input type="hidden" name ="table" value="' . $tableparam .'" />';
         $o.= '    <input type="hidden" name ="index" value="' . $indexparam .'" />';
         $o.= '    <input type="hidden" name ="sesskey" value="' . sesskey() .'" />';
@@ -139,18 +139,18 @@ class edit_index extends XMLDBAction {
         $b = ' <p class="centerpara buttons">';
         // The view original XML button
         if ($table->getIndex($indexparam)) {
-            $b .= '&nbsp;<a href="index.php?action=view_index_xml&amp;dir=' . urlencode(str_replace($CFG->dirroot, '', $dirpath)) . '&amp;select=original&amp;table=' . $tableparam . '&amp;index=' . $indexparam . '">[' . $this->str['vieworiginal'] . ']</a>';
+            $b .= '&nbsp;<a href="index.php?action=view_index_xml&amp;dir=' . urlencode(\core\component::to_mono_path($dirpath, true)) . '&amp;select=original&amp;table=' . $tableparam . '&amp;index=' . $indexparam . '">[' . $this->str['vieworiginal'] . ']</a>';
         } else {
             $b .= '&nbsp;[' . $this->str['vieworiginal'] . ']';
         }
         // The view edited XML button
         if ($index->hasChanged()) {
-            $b .= '&nbsp;<a href="index.php?action=view_index_xml&amp;dir=' . urlencode(str_replace($CFG->dirroot, '', $dirpath)) . '&amp;select=edited&amp;table=' . $tableparam . '&amp;index=' . $indexparam . '">[' . $this->str['viewedited'] . ']</a>';
+            $b .= '&nbsp;<a href="index.php?action=view_index_xml&amp;dir=' . urlencode(\core\component::to_mono_path($dirpath, true)) . '&amp;select=edited&amp;table=' . $tableparam . '&amp;index=' . $indexparam . '">[' . $this->str['viewedited'] . ']</a>';
         } else {
             $b .= '&nbsp;[' . $this->str['viewedited'] . ']';
         }
         // The back to edit table button
-        $b .= '&nbsp;<a href="index.php?action=edit_table&amp;table=' . $tableparam . '&amp;dir=' . urlencode(str_replace($CFG->dirroot, '', $dirpath)) . '">[' . $this->str['back'] . ']</a>';
+        $b .= '&nbsp;<a href="index.php?action=edit_table&amp;table=' . $tableparam . '&amp;dir=' . urlencode(\core\component::to_mono_path($dirpath, true)) . '">[' . $this->str['back'] . ']</a>';
         $b .= '</p>';
         $o .= $b;
 
