diff --git a/public/filter/algebra/algebradebug.php b/public/filter/algebra/algebradebug.php
index 264736dea29..53f7befacec 100644
--- a/public/filter/algebra/algebradebug.php
+++ b/public/filter/algebra/algebradebug.php
@@ -77,7 +77,6 @@
     }
 
 function algebra2tex($algebra) {
-  global $CFG;
   $algebra = str_replace('&lt;','<',$algebra);
   $algebra = str_replace('&gt;','>',$algebra);
   $algebra = str_replace('<>','#',$algebra);
@@ -95,22 +94,22 @@ function algebra2tex($algebra) {
   $algebra = escapeshellarg($algebra);
 
   if ( (PHP_OS == "WINNT") || (PHP_OS == "WIN32") || (PHP_OS == "Windows") ) {
-    $cmd  = "cd $CFG->dirroot\\filter\\algebra & algebra2tex.pl x/2";
+    $cmd  = "cd " . escapeshellarg(__DIR__) . " & algebra2tex.pl x/2";
     $test = `$cmd`;
     if ($test != '\frac{x}{2}') {
       echo "There is a problem with either Perl or the script algebra2tex.pl<br/>";
       $ecmd = $cmd . " 2>&1";
       echo `$ecmd` . "<br/>\n";
       echo "The shell command<br/>$cmd<br/>returned status = $status<br/>\n";
-      $commandpath = "$CFG->dirroot\\filter\\algebra\\algebra2tex.pl";
+      $commandpath = __DIR__ . '/algebra2tex.pl';
       if (file_exists($commandpath)) {
         echo "The file permissions of algebra2tex.pl are: " . decoct(fileperms($commandpath)) . "<br/>";
       }
       die;
     }
-    $cmd  = "cd $CFG->dirroot\\filter\\algebra & algebra2tex.pl $algebra";
+    $cmd  = "cd " . escapeshellarg(__DIR__) . " & algebra2tex.pl $algebra";
   } else {
-    $cmd  = "cd $CFG->dirroot/filter/algebra; ./algebra2tex.pl x/2";
+    $cmd  = "cd " . escapeshellarg(__DIR__) . "; ./algebra2tex.pl x/2";
     $test = `$cmd`;
     if ($test != '\frac{x}{2}') {
       echo "There is a problem with either Perl or the script algebra2tex.pl<br/>";
@@ -123,7 +122,7 @@ function algebra2tex($algebra) {
       }
       die;
     }
-    $cmd  = "cd $CFG->dirroot/filter/algebra; ./algebra2tex.pl $algebra";
+    $cmd  = "cd " . escapeshellarg(__DIR__) . "; ./algebra2tex.pl $algebra";
   }
   $texexp = `$cmd`;
   return $texexp;
