diff --git a/public/lib/classes/shutdown_manager.php b/public/lib/classes/shutdown_manager.php
index a64929b71ea..834509a50f2 100644
--- a/public/lib/classes/shutdown_manager.php
+++ b/public/lib/classes/shutdown_manager.php
@@ -280,7 +280,7 @@ class shutdown_manager {
                         $ts += $fs;
                         $hfs = display_size($fs);
                         // phpcs:ignore moodle.PHP.ForbiddenFunctions.FoundWithAlternative
-                        error_log(substr($f, strlen($CFG->dirroot)) . " size: $fs ($hfs)");
+                        error_log(\core\component::to_mono_path($f, leadingslash: true) . " size: $fs ($hfs)");
                     } else {
                         // phpcs:ignore moodle.PHP.ForbiddenFunctions.FoundWithAlternative
                         error_log($f);
