diff --git a/public/mod/h5pactivity/tests/generator/lib.php b/public/mod/h5pactivity/tests/generator/lib.php
index 2e15cff0582..dfc3fdbe174 100644
--- a/public/mod/h5pactivity/tests/generator/lib.php
+++ b/public/mod/h5pactivity/tests/generator/lib.php
@@ -55,7 +55,7 @@ class mod_h5pactivity_generator extends testing_module_generator {
         // Fill in optional values if not specified.
         if (!isset($record->packagefilepath)) {
             $record->packagefilepath = $CFG->dirroot.'/h5p/tests/fixtures/ipsums.h5p';
-        } else if (strpos($record->packagefilepath, $CFG->dirroot) !== 0) {
+        } else if (!\core\component::is_inside_codebase($record->packagefilepath, '')) {
             $record->packagefilepath = "{$CFG->dirroot}/{$record->packagefilepath}";
         }
         if (!isset($record->grade)) {
