diff --git a/public/admin/settings/plugins.php b/public/admin/settings/plugins.php
index 5e3a2169017..9898c3e39ac 100644
--- a/public/admin/settings/plugins.php
+++ b/public/admin/settings/plugins.php
@@ -394,7 +394,11 @@ if ($hassiteconfig) {
     $ADMIN->add('portfoliosettings', new admin_externalpage('portfoliocontroller', new lang_string('manageportfolios', 'portfolio'), $url, 'moodle/site:config', true));
 
     foreach (portfolio_instances(false, false) as $portfolio) {
-        require_once($CFG->dirroot . '/portfolio/' . $portfolio->get('plugin') . '/lib.php');
+        $plugindir = \core_component::get_plugin_directory('portfolio', $portfolio->get('plugin'));
+        if ($plugindir === null) {
+            throw new \coding_exception('Plugin not installed: portfolio_' . $portfolio->get('plugin'));
+        }
+        require_once($plugindir . '/lib.php');
         $classname = 'portfolio_plugin_' . $portfolio->get('plugin');
         $ADMIN->add(
             'portfoliosettings',
