diff --git a/public/auth/test_settings.php b/public/auth/test_settings.php
index 84207ba4a77..6d8cd33fe76 100644
--- a/public/auth/test_settings.php
+++ b/public/auth/test_settings.php
@@ -28,8 +28,11 @@ require_once("$CFG->libdir/adminlib.php");
 $auth = optional_param('auth', '', PARAM_RAW);
 if (!core_component::is_valid_plugin_name('auth', $auth)) {
     $auth = '';
-} else if (!file_exists("$CFG->dirroot/auth/$auth/auth.php")) {
-    $auth = '';
+} else {
+    $authdir = \core_component::get_plugin_directory('auth', $auth);
+    if (!$authdir || !file_exists("$authdir/auth.php")) {
+        $auth = '';
+    }
 }
 
 navigation_node::override_active_url(new moodle_url('/admin/settings.php', array('section'=>'manageauths')));
