diff --git a/public/course/classes/cm_info.php b/public/course/classes/cm_info.php
index de8ab6c35a0..695a2f9409b 100644
--- a/public/course/classes/cm_info.php
+++ b/public/course/classes/cm_info.php
@@ -1685,10 +1685,9 @@ class cm_info implements IteratorAggregate {
      * @return void
      */
     private function call_mod_function($type) {
-        global $CFG;
-        $libfile = $CFG->dirroot . '/mod/' . $this->modname . '/lib.php';
-        if (file_exists($libfile)) {
-            include_once($libfile);
+        $moddir = \core_component::get_plugin_directory('mod', $this->modname);
+        if ($moddir && file_exists($moddir . '/lib.php')) {
+            include_once($moddir . '/lib.php');
             $function = 'mod_' . $this->modname . '_' . $type;
             if (function_exists($function)) {
                 $function($this);
