diff --git a/public/course/classes/local/repository/content_item_readonly_repository.php b/public/course/classes/local/repository/content_item_readonly_repository.php
index cfd752fdacc..ee5cda16df9 100644
--- a/public/course/classes/local/repository/content_item_readonly_repository.php
+++ b/public/course/classes/local/repository/content_item_readonly_repository.php
@@ -150,7 +150,7 @@ class content_item_readonly_repository implements content_item_readonly_reposito
      * @return array the array of content items.
      */
     public function find_all(): array {
-        global $OUTPUT, $DB, $CFG;
+        global $OUTPUT, $DB;
 
         // Get all modules so we know which plugins are enabled and able to add content.
         // Only module plugins may add content items.
@@ -160,7 +160,8 @@ class content_item_readonly_repository implements content_item_readonly_reposito
         // Now, generate the content_items.
         foreach ($modules as $modid => $mod) {
             // Exclude modules if the code doesn't exist.
-            if (!file_exists("$CFG->dirroot/mod/$mod->name/lib.php")) {
+            $moddir = \core_component::get_plugin_directory('mod', $mod->name);
+            if (!$moddir || !file_exists("$moddir/lib.php")) {
                 continue;
             }
             // Create the content item for the module itself.
@@ -222,7 +223,7 @@ class content_item_readonly_repository implements content_item_readonly_reposito
      * @return array the array of content_item objects
      */
     public function find_all_for_course(\stdClass $course, \stdClass $user): array {
-        global $OUTPUT, $DB, $CFG;
+        global $OUTPUT, $DB;
 
         // Get all modules so we know which plugins are enabled and able to add content.
         // Only module plugins may add content items.
@@ -232,7 +233,8 @@ class content_item_readonly_repository implements content_item_readonly_reposito
         // Now, generate the content_items.
         foreach ($modules as $modid => $mod) {
             // Exclude modules if the code doesn't exist.
-            if (!file_exists("$CFG->dirroot/mod/$mod->name/lib.php")) {
+            $moddir = \core_component::get_plugin_directory('mod', $mod->name);
+            if (!$moddir || !file_exists("$moddir/lib.php")) {
                 continue;
             }
             // Create the content item for the module itself.
