diff --git a/public/course/classes/modinfo.php b/public/course/classes/modinfo.php
index 0997d016ba6..e3d5f026e44 100644
--- a/public/course/classes/modinfo.php
+++ b/public/course/classes/modinfo.php
@@ -697,7 +697,8 @@ class modinfo {
 
             // Skip modules which don't exist.
             if (!array_key_exists($mod->mod, $modexists)) {
-                $modexists[$mod->mod] = file_exists("$CFG->dirroot/mod/$mod->mod/lib.php");
+                $moddir = \core_component::get_plugin_directory('mod', $mod->mod);
+                $modexists[$mod->mod] = $moddir && file_exists("$moddir/lib.php");
             }
             if (!$modexists[$mod->mod]) {
                 continue;
@@ -1105,11 +1106,12 @@ class modinfo {
                         $modname = $mods[$cmid]->mod;
                         $functionname = $modname . "_get_coursemodule_info";
 
-                        if (!file_exists("$CFG->dirroot/mod/$modname/lib.php")) {
+                        $moddir = \core_component::get_plugin_directory('mod', $modname);
+                        if (!$moddir || !file_exists("$moddir/lib.php")) {
                             continue;
                         }
 
-                        include_once("$CFG->dirroot/mod/$modname/lib.php");
+                        include_once("$moddir/lib.php");
 
                         if ($hasfunction = function_exists($functionname)) {
                             if ($info = $functionname($rawmods[$cmid])) {
