diff --git a/public/course/reset_form.php b/public/course/reset_form.php
index c2d522a07fc..f418841c80c 100644
--- a/public/course/reset_form.php
+++ b/public/course/reset_form.php
@@ -32,7 +32,7 @@ class course_reset_form extends moodleform {
      * Form definition.
      */
     public function definition() {
-        global $CFG, $COURSE, $DB;
+        global $COURSE, $DB;
 
         $mform = $this->_form;
 
@@ -84,14 +84,14 @@ class course_reset_form extends moodleform {
         if ($allmods = $DB->get_records('modules') ) {
             foreach ($allmods as $mod) {
                 $modname = $mod->name;
-                $modfile = $CFG->dirroot."/mod/$modname/lib.php";
+                $moddir = \core_component::get_plugin_directory('mod', $modname);
                 $modresetcourseformdefinition = $modname.'_reset_course_form_definition';
                 $modresetuserdata = $modname.'_reset_userdata';
-                if (file_exists($modfile)) {
+                if ($moddir && file_exists("$moddir/lib.php")) {
                     if (!$DB->count_records($modname, ['course' => $COURSE->id])) {
                         continue; // Skip mods with no instances.
                     }
-                    include_once($modfile);
+                    include_once("$moddir/lib.php");
                     if (function_exists($modresetcourseformdefinition)) {
                         $modresetcourseformdefinition($mform);
                     } else if (!function_exists($modresetuserdata)) {
@@ -161,7 +161,7 @@ class course_reset_form extends moodleform {
      * Method to load default values for the reset course form.
      */
     public function load_defaults() {
-        global $CFG, $COURSE, $DB;
+        global $COURSE, $DB;
 
         $mform = $this->_form;
 
@@ -180,10 +180,10 @@ class course_reset_form extends moodleform {
         if ($allmods = $DB->get_records('modules') ) {
             foreach ($allmods as $mod) {
                 $modname = $mod->name;
-                $modfile = $CFG->dirroot."/mod/$modname/lib.php";
+                $moddir = \core_component::get_plugin_directory('mod', $modname);
                 $modresetcourseformdefaults = $modname.'_reset_course_form_defaults';
-                if (file_exists($modfile)) {
-                    @include_once($modfile);
+                if ($moddir && file_exists("$moddir/lib.php")) {
+                    @include_once("$moddir/lib.php");
                     if (function_exists($modresetcourseformdefaults)) {
                         if ($moddefs = $modresetcourseformdefaults($COURSE)) {
                             $defaults = $defaults + $moddefs;
