diff --git a/public/filter/manage.php b/public/filter/manage.php
index 8fc78ff66ea..fe82d7454ab 100644
--- a/public/filter/manage.php
+++ b/public/filter/manage.php
@@ -81,7 +81,7 @@ if ($forfilter) {
     if (!filter_has_local_settings($forfilter)) {
         throw new \moodle_exception('filterdoesnothavelocalconfig', 'error', $forfilter);
     }
-    require_once($CFG->dirroot . '/filter/' . $forfilter . '/filterlocalsettings.php');
+    require_once(\core_component::get_plugin_directory('filter', $forfilter) . '/filterlocalsettings.php');
     $formname = $forfilter . '_filter_local_settings_form';
     $settingsform = new $formname($CFG->wwwroot . '/filter/manage.php', $forfilter, $context);
     if ($settingsform->is_cancelled()) {
