diff --git a/public/grade/report/lib.php b/public/grade/report/lib.php
index 464a38cb1b1..21492b170ef 100644
--- a/public/grade/report/lib.php
+++ b/public/grade/report/lib.php
@@ -817,7 +817,7 @@ abstract class grade_report {
      * @return array Item types.
      */
     public function item_types(): array {
-        global $DB, $CFG;
+        global $DB;
 
         $modnames = [];
         $sql = "(SELECT gi.itemmodule
@@ -832,7 +832,8 @@ abstract class grade_report {
 
         $itemtypes = $DB->get_records_sql($sql, ['courseid1' => $this->courseid, 'courseid2' => $this->courseid]);
         foreach ($itemtypes as $itemtype => $value) {
-            if (file_exists("$CFG->dirroot/mod/$itemtype/lib.php")) {
+            $moddir = \core_component::get_plugin_directory('mod', $itemtype);
+            if ($moddir && file_exists("$moddir/lib.php")) {
                 $modnames[$itemtype] = get_string("modulename", $itemtype, null, true);
             } else if ($itemtype == 'manual') {
                 $modnames[$itemtype] = get_string('manualitem', 'grades', null, true);
