diff --git a/public/grade/grading/lib.php b/public/grade/grading/lib.php
index 743020d6094..f87fd3942e8 100644
--- a/public/grade/grading/lib.php
+++ b/public/grade/grading/lib.php
@@ -491,7 +491,7 @@ class grading_manager {
      * @return gradingform_controller
      */
     public function get_controller($method) {
-        global $CFG, $DB;
+        global $DB;
 
         $this->ensure_isset(array('context', 'component', 'area'));
 
@@ -524,7 +524,7 @@ class grading_manager {
             $this->areacache = $DB->get_record('grading_areas', array('id' => $areaid), '*', MUST_EXIST);
         }
 
-        require_once($CFG->dirroot.'/grade/grading/form/'.$method.'/lib.php');
+        require_once(\core_component::get_plugin_directory('gradingform', $method) . '/lib.php');
         $classname = 'gradingform_'.$method.'_controller';
 
         return new $classname($this->context, $this->component, $this->area, $this->areacache->id);
diff --git a/public/lib/classes/grading_external.php b/public/lib/classes/grading_external.php
index b374e996c7f..835cf440821 100644
--- a/public/lib/classes/grading_external.php
+++ b/public/lib/classes/grading_external.php
@@ -247,7 +247,6 @@ class core_grading_external extends external_api {
      * @since  Moodle 2.5
      */
     private static function definition() {
-        global $CFG;
         $definition = array();
         $definition['id']                = new external_value(PARAM_INT, 'definition id', VALUE_OPTIONAL);
         $definition['method']            = new external_value(PARAM_TEXT, 'method');
@@ -262,7 +261,7 @@ class core_grading_external extends external_api {
         $definition['usermodified']      = new external_value(PARAM_INT, 'user who modified definition');
         $definition['timecopied']        = new external_value(PARAM_INT, 'time copied', VALUE_OPTIONAL);
         foreach (self::get_grading_methods() as $method) {
-            require_once($CFG->dirroot.'/grade/grading/form/'.$method.'/lib.php');
+            require_once(\core_component::get_plugin_directory('gradingform', $method) . '/lib.php');
             $details  = call_user_func('gradingform_'.$method.'_controller::get_external_definition_details');
             if ($details != null) {
                 $items = array();
@@ -412,7 +411,6 @@ class core_grading_external extends external_api {
      * @since  Moodle 2.6
      */
     private static function grading_instance() {
-        global $CFG;
         $instance = array();
         $instance['id']                = new external_value(PARAM_INT, 'instance id');
         $instance['raterid']           = new external_value(PARAM_INT, 'rater id');
@@ -423,7 +421,7 @@ class core_grading_external extends external_api {
         $instance['feedbackformat']    = new external_format_value('feedback', VALUE_OPTIONAL);
         $instance['timemodified']      = new external_value(PARAM_INT, 'modified time');
         foreach (self::get_grading_methods() as $method) {
-            require_once($CFG->dirroot.'/grade/grading/form/'.$method.'/lib.php');
+            require_once(\core_component::get_plugin_directory('gradingform', $method) . '/lib.php');
             $details  = call_user_func('gradingform_'.$method.'_controller::get_external_instance_filling_details');
             if ($details != null) {
                 $items = array();
@@ -535,7 +533,7 @@ class core_grading_external extends external_api {
         $definitionobject->description_editor = array('text' => $text, 'format' => $format);
 
         require_once("$CFG->libdir/filelib.php");
-        require_once($CFG->dirroot.'/grade/grading/form/'.$method.'/lib.php');
+        require_once(\core_component::get_plugin_directory('gradingform', $method) . '/lib.php');
         $details  = call_user_func('gradingform_'.$method.'_controller::get_external_definition_details');
         $methodarray = array();
         foreach (array_keys($details) as $definitionkey) {
