diff --git a/public/lib/filterlib.php b/public/lib/filterlib.php
index a56d203333d..27288e2c853 100644
--- a/public/lib/filterlib.php
+++ b/public/lib/filterlib.php
@@ -801,13 +801,14 @@ function filter_delete_all_for_context($contextid) {
  * @return boolean Whether there should be a 'Settings' link on the config page.
  */
 function filter_has_global_settings($filter) {
-    global $CFG;
-    $settingspath = $CFG->dirroot . '/filter/' . $filter . '/settings.php';
-    if (is_readable($settingspath)) {
+    $filterdir = \core_component::get_plugin_directory('filter', $filter);
+    if (!$filterdir) {
+        return false;
+    }
+    if (is_readable("$filterdir/settings.php")) {
         return true;
     }
-    $settingspath = $CFG->dirroot . '/filter/' . $filter . '/filtersettings.php';
-    return is_readable($settingspath);
+    return is_readable("$filterdir/filtersettings.php");
 }
 
 /**
@@ -817,9 +818,8 @@ function filter_has_global_settings($filter) {
  * @return boolean Whether there should be a 'Settings' link on the manage filters in context page.
  */
 function filter_has_local_settings($filter) {
-    global $CFG;
-    $settingspath = $CFG->dirroot . '/filter/' . $filter . '/filterlocalsettings.php';
-    return is_readable($settingspath);
+    $filterdir = \core_component::get_plugin_directory('filter', $filter);
+    return (bool) ($filterdir && is_readable("$filterdir/filterlocalsettings.php"));
 }
 
 /**
