diff --git a/public/lib/questionlib.php b/public/lib/questionlib.php
index ac5ffd7384c..7b61070f75a 100644
--- a/public/lib/questionlib.php
+++ b/public/lib/questionlib.php
@@ -1687,7 +1687,11 @@ function question_pluginfile($course, $context, $component, $filearea, $args, $f
         // Load parent class for import/export.
         require_once($CFG->dirroot . '/question/format.php');
         require_once($CFG->dirroot . '/question/editlib.php');
-        require_once($CFG->dirroot . '/question/format/' . $format . '/format.php');
+        $formatdir = \core_component::get_plugin_directory('qformat', $format);
+        if ($formatdir === null) {
+            send_file_not_found();
+        }
+        require_once("{$formatdir}/format.php");
 
         $classname = 'qformat_' . $format;
         if (!class_exists($classname)) {
