diff --git a/public/lib/portfolio/forms.php b/public/lib/portfolio/forms.php
index 718757b960a..8977ea84ac3 100644
--- a/public/lib/portfolio/forms.php
+++ b/public/lib/portfolio/forms.php
@@ -197,7 +197,7 @@ final class portfolio_admin_form extends moodleform {
         // let the plugin add the fields they want (either statically or not)
         if (portfolio_static_function($this->plugin, 'has_admin_config')) {
             require_once($CFG->libdir . '/portfolio/plugin.php');
-            require_once($CFG->dirroot . '/portfolio/' . $this->plugin .  '/lib.php');
+            require_once(\core_component::get_plugin_directory('portfolio', $this->plugin) . '/lib.php');
             $classname = 'portfolio_plugin_' . $this->plugin;
             $classname::admin_config_form($mform);
         }
