diff --git a/public/lib/portfolio/plugin.php b/public/lib/portfolio/plugin.php
index a387f30357f..d02adceff9e 100644
--- a/public/lib/portfolio/plugin.php
+++ b/public/lib/portfolio/plugin.php
@@ -421,7 +421,7 @@ abstract class portfolio_plugin_base {
      * @return object subclass of portfolio_plugin_base
      */
     public static function create_instance($plugin, $name, $config) {
-        global $DB, $CFG;
+        global $DB;
         $new = (object)array(
             'plugin' => $plugin,
             'name'   => $name,
@@ -433,7 +433,11 @@ abstract class portfolio_plugin_base {
             }
         }
         $newid = $DB->insert_record('portfolio_instance', $new);
-        require_once($CFG->dirroot . '/portfolio/' . $plugin . '/lib.php');
+        $plugindir = \core_component::get_plugin_directory('portfolio', $plugin);
+        if ($plugindir === null) {
+            throw new \coding_exception("Plugin not installed: portfolio_$plugin");
+        }
+        require_once($plugindir . '/lib.php');
         $classname = 'portfolio_plugin_'  . $plugin;
         $obj = new $classname($newid);
         $obj->set_config($config);
