diff --git a/public/lib/portfoliolib.php b/public/lib/portfoliolib.php
index 2769d2080ac..653c2562dfa 100644
--- a/public/lib/portfoliolib.php
+++ b/public/lib/portfoliolib.php
@@ -785,7 +785,11 @@ function portfolio_instance($instanceid, $record=null) {
         }
     }
     require_once($CFG->libdir . '/portfolio/plugin.php');
-    require_once($CFG->dirroot . '/portfolio/'. $instance->plugin . '/lib.php');
+    $plugindir = \core_component::get_plugin_directory('portfolio', $instance->plugin);
+    if ($plugindir === null) {
+        throw new \coding_exception("Plugin not installed: portfolio_{$instance->plugin}");
+    }
+    require_once($plugindir . '/lib.php');
     $classname = 'portfolio_plugin_' . $instance->plugin;
     return new $classname($instanceid, $instance);
 }
@@ -821,7 +825,11 @@ function portfolio_static_function($plugin, $function) {
     }
 
     require_once($CFG->libdir . '/portfolio/plugin.php');
-    require_once($CFG->dirroot . '/portfolio/' . $plugin .  '/lib.php');
+    $plugindir = \core_component::get_plugin_directory('portfolio', $pname);
+    if ($plugindir === null) {
+        throw new \coding_exception("Plugin not installed: portfolio_$pname");
+    }
+    require_once($plugindir . '/lib.php');
     return call_user_func_array(array('portfolio_plugin_' . $plugin, $function), $args);
 }
 
