diff --git a/public/message/classes/api.php b/public/message/classes/api.php
index f0196f8ecbf..1b8c6e05a46 100644
--- a/public/message/classes/api.php
+++ b/public/message/classes/api.php
@@ -1736,10 +1736,11 @@ class api {
      * @since Moodle 3.2
      */
     public static function get_processed_processor_object(\stdClass $processor) {
-        global $CFG;
-
-        $processorfile = $CFG->dirroot. '/message/output/'.$processor->name.'/message_output_'.$processor->name.'.php';
-        if (is_readable($processorfile)) {
+        global $CFG; // Needed in scope for included processor files.
+
+        $processordir = \core_component::get_plugin_directory('message', $processor->name);
+        $processorfile = $processordir ? $processordir . '/message_output_' . $processor->name . '.php' : null;
+        if ($processorfile && is_readable($processorfile)) {
             include_once($processorfile);
             $processclass = 'message_output_' . $processor->name;
             if (class_exists($processclass)) {
@@ -1750,7 +1751,7 @@ class api {
                     $processor->configured = 1;
                 }
                 $processor->hassettings = 0;
-                if (is_readable($CFG->dirroot.'/message/output/'.$processor->name.'/settings.php')) {
+                if (is_readable($processordir . '/settings.php')) {
                     $processor->hassettings = 1;
                 }
                 $processor->available = 1;
