diff --git a/public/message/lib.php b/public/message/lib.php
index 9d52400bcb7..8929d688293 100644
--- a/public/message/lib.php
+++ b/public/message/lib.php
@@ -410,14 +410,15 @@ function get_message_providers() {
  * @return message_output message_output the requested class.
  */
 function get_message_processor($type) {
-    global $CFG;
+    global $CFG; // Needed in scope for included processor files.
 
     // Note, we cannot use the get_message_processors function here, becaues this
     // code is called during install after installing each messaging plugin, and
     // get_message_processors caches the list of installed plugins.
 
-    $processorfile = $CFG->dirroot . "/message/output/{$type}/message_output_{$type}.php";
-    if (!is_readable($processorfile)) {
+    $processordir = \core_component::get_plugin_directory('message', $type);
+    $processorfile = $processordir ? $processordir . "/message_output_{$type}.php" : null;
+    if (!$processorfile || !is_readable($processorfile)) {
         throw new coding_exception('Unknown message processor type ' . $type);
     }
 
