diff --git a/public/mod/lesson/import.php b/public/mod/lesson/import.php
index ec15da8f765..975190d60e1 100644
--- a/public/mod/lesson/import.php
+++ b/public/mod/lesson/import.php
@@ -77,11 +77,11 @@ if ($data = $mform->get_data()) {
     }
 
     $formatclass = 'qformat_'.$data->format;
-    $formatclassfile = $CFG->dirroot.'/question/format/'.$data->format.'/format.php';
-    if (!is_readable($formatclassfile)) {
+    $qformatdir = \core_component::get_plugin_directory('qformat', $data->format);
+    if (!$qformatdir || !is_readable("$qformatdir/format.php")) {
         throw new \moodle_exception('unknowformat', '', '', $data->format);
     }
-    require_once($formatclassfile);
+    require_once("$qformatdir/format.php");
     $format = new $formatclass();
 
     $format->set_importcontext($context);
