diff --git a/public/mod/assign/externallib.php b/public/mod/assign/externallib.php
index 166d7b64e16..7e7441db10f 100644
--- a/public/mod/assign/externallib.php
+++ b/public/mod/assign/externallib.php
@@ -1907,7 +1907,11 @@ class mod_assign_external extends \mod_assign\external\external_api {
         $advancedgradingdata = array();
         $methods = array_keys(grading_manager::available_methods(false));
         foreach ($methods as $method) {
-            require_once($CFG->dirroot.'/grade/grading/form/'.$method.'/lib.php');
+            $gradingformdir = \core_component::get_plugin_directory('gradingform', $method);
+            if ($gradingformdir === null) {
+                throw new \coding_exception("Plugin not installed: gradingform_{$method}");
+            }
+            require_once("$gradingformdir/lib.php");
             $details  = call_user_func('gradingform_'.$method.'_controller::get_external_instance_filling_details');
             if (!empty($details)) {
                 $items = array();
@@ -2043,7 +2047,11 @@ class mod_assign_external extends \mod_assign\external\external_api {
         $advancedgradingdata = array();
         $methods = array_keys(grading_manager::available_methods(false));
         foreach ($methods as $method) {
-            require_once($CFG->dirroot.'/grade/grading/form/'.$method.'/lib.php');
+            $gradingformdir = \core_component::get_plugin_directory('gradingform', $method);
+            if ($gradingformdir === null) {
+                throw new \coding_exception("Plugin not installed: gradingform_{$method}");
+            }
+            require_once("$gradingformdir/lib.php");
             $details  = call_user_func('gradingform_'.$method.'_controller::get_external_instance_filling_details');
             if (!empty($details)) {
                 $items = array();
