diff --git a/public/mod/workshop/locallib.php b/public/mod/workshop/locallib.php
index 5d7a22878f0..45d0f83a69c 100644
--- a/public/mod/workshop/locallib.php
+++ b/public/mod/workshop/locallib.php
@@ -1609,11 +1609,12 @@ class workshop {
             if (empty($this->strategy)) {
                 throw new coding_exception('Unknown grading strategy');
             }
-            $strategylib = __DIR__ . '/form/' . $this->strategy . '/lib.php';
-            if (is_readable($strategylib)) {
+            $strategydir = \core_component::get_plugin_directory('workshopform', $this->strategy);
+            $strategylib = $strategydir ? $strategydir . '/lib.php' : null;
+            if ($strategylib && is_readable($strategylib)) {
                 require_once($strategylib);
             } else {
-                throw new coding_exception('the grading forms subplugin must contain library ' . $strategylib);
+                throw new coding_exception('the grading forms subplugin workshopform_' . $this->strategy . ' must contain library lib.php');
             }
             $classname = 'workshop_' . $this->strategy . '_strategy';
             $this->strategyinstance = new $classname($this);
@@ -1635,9 +1636,10 @@ class workshop {
         global $DB;
 
         $method = clean_param($method, PARAM_PLUGIN);
-        $evaluationlib = __DIR__ . '/eval/' . $method . '/lib.php';
+        $evaluationdir = \core_component::get_plugin_directory('workshopeval', $method);
+        $evaluationlib = $evaluationdir ? $evaluationdir . '/lib.php' : null;
 
-        if (is_readable($evaluationlib)) {
+        if ($evaluationlib && is_readable($evaluationlib)) {
             $this->evaluationinstance = null;
             $this->evaluation = $method;
             $DB->set_field('workshop', 'evaluation', $method, array('id' => $this->id));
@@ -1659,18 +1661,20 @@ class workshop {
             if (empty($this->evaluation)) {
                 $this->evaluation = 'best';
             }
-            $evaluationlib = __DIR__ . '/eval/' . $this->evaluation . '/lib.php';
-            if (is_readable($evaluationlib)) {
+            $evaluationdir = \core_component::get_plugin_directory('workshopeval', $this->evaluation);
+            $evaluationlib = $evaluationdir ? $evaluationdir . '/lib.php' : null;
+            if ($evaluationlib && is_readable($evaluationlib)) {
                 require_once($evaluationlib);
             } else {
                 // Fall back in case the subplugin is not available.
                 $this->evaluation = 'best';
-                $evaluationlib = __DIR__ . '/eval/' . $this->evaluation . '/lib.php';
-                if (is_readable($evaluationlib)) {
+                $evaluationdir = \core_component::get_plugin_directory('workshopeval', $this->evaluation);
+                $evaluationlib = $evaluationdir ? $evaluationdir . '/lib.php' : null;
+                if ($evaluationlib && is_readable($evaluationlib)) {
                     require_once($evaluationlib);
                 } else {
                     // Fall back in case the subplugin is not available any more.
-                    throw new coding_exception('Missing default grading evaluation library ' . $evaluationlib);
+                    throw new coding_exception('Missing default grading evaluation library workshopeval_' . $this->evaluation);
                 }
             }
             $classname = 'workshop_' . $this->evaluation . '_evaluation';
@@ -1691,11 +1695,12 @@ class workshop {
     public function allocator_instance($method) {
         global $CFG;    // because we require other libs here
 
-        $allocationlib = __DIR__ . '/allocation/' . $method . '/lib.php';
-        if (is_readable($allocationlib)) {
+        $allocationdir = \core_component::get_plugin_directory('workshopallocation', $method);
+        $allocationlib = $allocationdir ? $allocationdir . '/lib.php' : null;
+        if ($allocationlib && is_readable($allocationlib)) {
             require_once($allocationlib);
         } else {
-            throw new coding_exception('Unable to find the allocation library ' . $allocationlib);
+            throw new coding_exception('Unable to find the allocation library workshopallocation_' . $method);
         }
         $classname = 'workshop_' . $method . '_allocator';
         return new $classname($this);
