diff --git a/public/mod/quiz/report.php b/public/mod/quiz/report.php
index 2b9f31fff56..43e5046b557 100644
--- a/public/mod/quiz/report.php
+++ b/public/mod/quiz/report.php
@@ -66,14 +66,13 @@ if ($mode == '') {
 } else if (!in_array($mode, $reportlist)) {
     throw new \moodle_exception('erroraccessingreport', 'quiz');
 }
-if (!is_readable("report/$mode/report.php")) {
-    throw new \moodle_exception('reportnotfound', 'quiz', '', $mode);
-}
 
 // Open the selected quiz report and display it.
-$file = $CFG->dirroot . '/mod/quiz/report/' . $mode . '/report.php';
-if (is_readable($file)) {
-    include_once($file);
+$quizreportdir = \core_component::get_plugin_directory('quiz', $mode);
+if ($quizreportdir && is_readable("$quizreportdir/report.php")) {
+    include_once("$quizreportdir/report.php");
+} else {
+    throw new \moodle_exception('reportnotfound', 'quiz', '', $mode);
 }
 $reportclassname = 'quiz_' . $mode . '_report';
 if (!class_exists($reportclassname)) {
