diff --git a/public/question/bank/importquestions/classes/form/question_import_form.php b/public/question/bank/importquestions/classes/form/question_import_form.php
index d6d7a912992..dee1c55d8e0 100644
--- a/public/question/bank/importquestions/classes/form/question_import_form.php
+++ b/public/question/bank/importquestions/classes/form/question_import_form.php
@@ -124,7 +124,6 @@ class question_import_form extends moodleform {
      * @throws moodle_exception
      */
     protected function validate_uploaded_file($data, $errors) {
-        global $CFG;
 
         if (empty($data['newfile'])) {
             $errors['newfile'] = get_string('required');
@@ -142,12 +141,12 @@ class question_import_form extends moodleform {
             return $errors;
         }
 
-        $formatfile = $CFG->dirroot . '/question/format/' . $data['format'] . '/format.php';
-        if (!is_readable($formatfile)) {
+        $qformatdir = \core_component::get_plugin_directory('qformat', $data['format']);
+        if (!$qformatdir || !is_readable("$qformatdir/format.php")) {
             throw new moodle_exception('formatnotfound', 'question', '', $data['format']);
         }
 
-        require_once($formatfile);
+        require_once("$qformatdir/format.php");
 
         $classname = 'qformat_' . $data['format'];
         $qformat = new $classname();
