diff --git a/public/question/bank/exportquestions/export.php b/public/question/bank/exportquestions/export.php
index ca08cc60a13..98abbea5953 100644
--- a/public/question/bank/exportquestions/export.php
+++ b/public/question/bank/exportquestions/export.php
@@ -61,7 +61,8 @@ $exportform = new export_form($thispageurl,
 
 if ($fromform = $exportform->get_data()) {
     $thiscontext = $contexts->lowest();
-    if (!is_readable($CFG->dirroot . "/question/format/{$fromform->format}/format.php")) {
+    $qformatdir = \core_component::get_plugin_directory('qformat', $fromform->format);
+    if (!$qformatdir || !is_readable("$qformatdir/format.php")) {
         throw new moodle_exception('unknowformat', '', '', $fromform->format);
     }
     $withcategories = 'nocategories';
