diff --git a/public/question/classes/privacy/provider.php b/public/question/classes/privacy/provider.php
index 8a9857f4791..d4d67814a2a 100644
--- a/public/question/classes/privacy/provider.php
+++ b/public/question/classes/privacy/provider.php
@@ -436,7 +436,7 @@ class provider implements
      * @param   approved_contextlist    $contextlist    The approved contexts to export information for.
      */
     public static function export_user_data(approved_contextlist $contextlist) {
-        global $CFG, $DB, $SITE;
+        global $DB, $SITE;
         if (empty($contextlist)) {
             return;
         }
@@ -444,7 +444,11 @@ class provider implements
         // Use the Moodle XML Data format.
         // It is the only lossless format that we support.
         $format = "xml";
-        require_once($CFG->dirroot . "/question/format/{$format}/format.php");
+        $qformatdir = \core_component::get_plugin_directory('qformat', $format);
+        if ($qformatdir === null) {
+            throw new \coding_exception("Plugin not installed: qformat_{$format}");
+        }
+        require_once("$qformatdir/format.php");
 
         // THe export system needs questions in a particular format.
         // The easiest way to fetch these is with get_questions_category() which takes the details of a question
