diff --git a/public/question/type/questiontypebase.php b/public/question/type/questiontypebase.php
index 95d6ec29cc8..4426ad9c76f 100644
--- a/public/question/type/questiontypebase.php
+++ b/public/question/type/questiontypebase.php
@@ -202,8 +202,7 @@ class question_type {
             $contexts, $formeditable) {
         global $CFG;
         require_once($CFG->dirroot . '/question/type/edit_question_form.php');
-        $definitionfile = $CFG->dirroot . '/question/type/' . $this->name() .
-                '/edit_' . $this->name() . '_form.php';
+        $definitionfile = $this->plugin_dir() . '/edit_' . $this->name() . '_form.php';
         if (!is_readable($definitionfile) || !is_file($definitionfile)) {
             throw new coding_exception($this->plugin_name() .
                     ' is missing the definition of its editing formin file ' .
@@ -224,6 +223,10 @@ class question_type {
      */
     public function plugin_dir() {
         global $CFG;
+        $dir = \core_component::get_plugin_directory('qtype', $this->name());
+        if ($dir !== null) {
+            return $dir;
+        }
         return $CFG->dirroot . '/question/type/' . $this->name();
     }
 
