diff --git a/public/rating/lib.php b/public/rating/lib.php
index 2da1494669d..4eb430dbbe5 100644
--- a/public/rating/lib.php
+++ b/public/rating/lib.php
@@ -1146,7 +1146,11 @@ class rating_manager {
             if ($modinstance) {
                 $modinstance->cmidnumber = $cm->idnumber;
                 $functionname = $cm->modname.'_update_grades';
-                require_once($CFG->dirroot."/mod/{$cm->modname}/lib.php");
+                $moddir = \core_component::get_plugin_directory('mod', $cm->modname);
+                if ($moddir === null) {
+                    throw new \coding_exception("Plugin not installed: mod_{$cm->modname}");
+                }
+                require_once($moddir . '/lib.php');
                 if (function_exists($functionname)) {
                     $functionname($modinstance, $rateduserid);
                 }
