diff --git a/public/theme/styles_debug.php b/public/theme/styles_debug.php
index 83bb7c394f7..dc49aece024 100644
--- a/public/theme/styles_debug.php
+++ b/public/theme/styles_debug.php
@@ -38,11 +38,7 @@ $sheet     = optional_param('sheet', '', PARAM_SAFEDIR);
 $usesvg    = optional_param('svg', 1, PARAM_BOOL);
 $rtl       = optional_param('rtl', false, PARAM_BOOL);
 
-if (file_exists("$CFG->dirroot/theme/$themename/config.php")) {
-    // The theme exists in standard location - ok.
-} else if (!empty($CFG->themedir) and file_exists("$CFG->themedir/$themename/config.php")) {
-    // Alternative theme location contains this theme - ok.
-} else {
+if (\core_component::get_plugin_directory('theme', $themename) === null) {
     css_send_css_not_found();
 }
 
