diff --git a/public/theme/javascript.php b/public/theme/javascript.php
index 203ca94c08b..cdb53848c29 100644
--- a/public/theme/javascript.php
+++ b/public/theme/javascript.php
@@ -60,9 +60,8 @@ if ($type !== 'head' and $type !== 'footer') {
     die('Theme was not found, sorry.');
 }
 
-if (file_exists("$CFG->dirroot/theme/$themename/config.php")) {
-    // exists
-} else if (!empty($CFG->themedir) and file_exists("$CFG->themedir/$themename/config.php")) {
+$themedir = \core_component::get_plugin_directory('theme', $themename);
+if ($themedir && file_exists("$themedir/config.php")) {
     // exists
 } else {
     header('HTTP/1.0 404 not found');
