diff --git a/public/user/tests/profilelib_test.php b/public/user/tests/profilelib_test.php
index 9029dbe414c..6c4eb0361c0 100644
--- a/public/user/tests/profilelib_test.php
+++ b/public/user/tests/profilelib_test.php
@@ -90,8 +90,11 @@ final class profilelib_test extends \advanced_testcase {
         require_once($CFG->dirroot . '/user/profile/definelib.php');
         $datatypes = profile_list_datatypes();
         foreach ($datatypes as $datatype => $datatypename) {
-            require_once($CFG->dirroot . '/user/profile/field/' .
-                $datatype . '/field.class.php');
+            $fielddir = \core_component::get_plugin_directory('profilefield', $datatype);
+            if ($fielddir === null) {
+                throw new \coding_exception("Plugin not installed: profilefield_$datatype");
+            }
+            require_once($fielddir . '/field.class.php');
             $newfield = 'profile_field_' . $datatype;
             $formfield = new $newfield();
             $this->assertNotNull($formfield);
