diff --git a/public/user/classes/form/profile_field_form.php b/public/user/classes/form/profile_field_form.php
index f147cb9be99..4d17f3a5084 100644
--- a/public/user/classes/form/profile_field_form.php
+++ b/public/user/classes/form/profile_field_form.php
@@ -46,7 +46,11 @@ class profile_field_form extends dynamic_form {
 
         // Everything else is dependant on the data type.
         $datatype = $this->get_field_record()->datatype;
-        require_once($CFG->dirroot.'/user/profile/field/'.$datatype.'/define.class.php');
+        $fielddir = \core_component::get_plugin_directory('profilefield', $datatype);
+        if ($fielddir === null) {
+            throw new \coding_exception("Plugin not installed: profilefield_$datatype");
+        }
+        require_once("{$fielddir}/define.class.php");
         $newfield = 'profile_define_'.$datatype;
         $this->field = new $newfield();
 
