diff --git a/public/user/profile/lib.php b/public/user/profile/lib.php
index 831a7c2d610..0e82901c83e 100644
--- a/public/user/profile/lib.php
+++ b/public/user/profile/lib.php
@@ -623,9 +623,11 @@ class profile_field_base {
  * @return profile_field_base
  */
 function profile_get_user_field(string $type, int $fieldid = 0, int $userid = 0, ?stdClass $fielddata = null): profile_field_base {
-    global $CFG;
-
-    require_once("{$CFG->dirroot}/user/profile/field/{$type}/field.class.php");
+    $fielddir = \core_component::get_plugin_directory('profilefield', $type);
+    if ($fielddir === null) {
+        throw new \coding_exception("Plugin not installed: profilefield_$type");
+    }
+    require_once("{$fielddir}/field.class.php");
 
     // Return instance of profile field type.
     $profilefieldtype = "profile_field_{$type}";
